x/evm/keeper/keeper.go

```diff
@@ -36,7 +36,11 @@ type Keeper struct {
 	bankKeeper    types.BankKeeper
 	stakingKeeper types.StakingKeeper
 
+	// Store current context, only works in sequential execution
 	ctx sdk.Context
+	// Store current eth tx, used by `StateDb` methods during execution of `ApplyTransaction`
+	currentTx *ethtypes.Transaction
```

Comment:
I'd like to identify the core issue instead of storing the tx locally on a variable when possible

Reply:
I just pushed a version that store `txDecoder`, which is better than this I think.